docker/dev_env/run.sh

```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env bash
+
+set -e
+
+# Kudos https://stackoverflow.com/a/10910180 for macOS group ID reading
+case "$OSTYPE" in
+linux*)
+  host_docker_gid="$(getent group docker | cut -d: -f3)"
+  ;;
+
+darwin*)
+  host_docker_gid="$(dscl . -read /Groups/docker | awk '($1 == "PrimaryGroupID:") { print $2 }')"
+  ;;
+
+*)
+  printf "'%s' is not supported for Openverse development. Only Linux and macOS are supported. Use WSL if on Windows." "$OSTYPE" >>/dev/stderr
+  exit 1
+  ;;
+esac
+
+suppress_output() {
+  "$@" 2>/dev/null 1>/dev/null
+}
+
+opener_home="/home/opener"
+
+run_args=(
+  -i
+  --rm
+  --name openverse-dev-env
+  --env "OPENVERSE_PROJECT=$OPENVERSE_PROJECT"
+  --env "TERM=xterm-256color"
+  --user "$UID:$host_docker_gid"
+  --network host
+  # Bind the repo to the same exact location inside the container so that pre-commit
+  # and others don't get confused about where files are supposed to be
+  -v "$OPENVERSE_PROJECT:$OPENVERSE_PROJECT:rw,z"
+  --workdir "$OPENVERSE_PROJECT"
+  # Save the home directory of the container so we can reuse it each time
+  --mount "type=volume,src=openverse-dev-env,target=$opener_home"
+  # Expose the host's docker socket to the container so the container can run docker/compose etc
+  -v /var/run/docker.sock:/var/run/docker.sock
```

Comment:
This is genius. Before I read this, I was thinking this would be a Docker-in-Docker type scenario but this is far simpler and so much more sensible.

Reply:
I learned about it from https://distrobox.it/, which is pretty nifty (and a similar tool comes installed by default on Fedora, or at least the immutable versions)